jkube-kit/config/service/src/main/java/org/eclipse/jkube/kit/config/service/ApplyService.java

```diff
@@ -1361,6 +1374,15 @@ private void applyStandardEntities(String fileName, List<HasMetadata> entities)
         }
     }
 
+    public  void initializeControllerDetectors(IngressControllerDetectorService ingressControllerService) {
+        List<IngressControllerDetector> ingressControllerDetectors = new ArrayList<>();
+        ingressControllerDetectors.add(new NginxIngressControllerDetector(kubernetesClient));
+        ingressControllerDetectors.add(new ContourIngressControllerDetector(kubernetesClient));
+        ingressControllerDetectors.add(new KongIngressControllerDetector(kubernetesClient));
+        ingressControllerDetectors.add(new OpenShiftIngressControllerDetector(kubernetesClient));
+        ingressControllerService.setIngressControllerDetectors(ingressControllerDetectors);
```

Comment:
We want to use SPI (or our own _improved_ version - seen Enrichers and Generators) for this.